unittest/finite-differences.cpp

```diff
@@ -229,14 +306,18 @@ struct FiniteDiffJoint
     typedef typename JointModel::TangentVector_t TV;
     typedef typename LieGroup<JointModel>::type LieGroupType;
 
-    JointModel jmodel = init<JointModel>::run();
+    JmodelWithBounds jmodelWithBounds = init<JointModel>::run();
```

Comment:
```suggestion
    auto jmodelWithBounds = init<JointModel>::run();
```
Some compiler didn't like the missing template argument in the return type.